library/ssl_srv.c

```diff
@@ -4643,13 +4643,16 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     }
 
     ret = mbedtls_pk_verify( peer_pk,
-                                      md_alg, hash_start, hashlen,
-                                      ssl->in_msg + i, sig_len );
+                             md_alg, hash_start, hashlen,
+                             ssl->in_msg + i, sig_len );
 
     if( ret == 0 )
     {
         mbedtls_platform_random_delay();
 
+        ret = mbedtls_pk_verify( peer_pk,
```

Comment:
@AndrzejKurek I still feel that we have a weak point here.  Let's add additional return code for the second verify and return success only in case both return codes are true. In this case we will know that both were not skipped. The same on the client side.

Reply:
Added!